packages/core/test/errors.test.ts

```diff
@@ -40,18 +42,102 @@ describe('error handling', () => {
 
     installGlobalOnErrorHandler((ev) => {
       ev.preventDefault();
-      expect(ev.error.message).toEqual('blah');
+      expect(ev.error.message).toEqual(
+        'no_infinite_loop_when_error_is_thrown_in_subscribe'
+      );
       done();
     });
   });
 
-  it('unhandled rejections should be reported globally', (done) => {
-    const promiseMachine = createMachine({
+  it(`doesn't crash the actor when an error is thrown in subscribe`, (done) => {
+    const machine = createMachine({
+      id: 'machine',
+      initial: 'initial',
+      context: {
+        count: 0
+      },
+      states: {
+        initial: {
+          on: { activate: 'active' }
+        },
+        active: {}
+      }
+    });
+
+    const spy = jest.fn().mockImplementation(() => {
+      throw new Error('doesnt_crash_actor_when_error_is_thrown_in_subscribe');
+    });
+
+    const actor = interpret(machine).start();
+
+    actor.subscribe(spy);
+    actor.send({ type: 'activate' });
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(actor.status).toEqual(ActorStatus.Running);
+
+    installGlobalOnErrorHandler((ev) => {
+      ev.preventDefault();
+      expect(ev.error.message).toEqual(
+        'doesnt_crash_actor_when_error_is_thrown_in_subscribe'
+      );
+      done();
+    });
+  });
+
+  it(`doesn't notify error listener when an error is thrown in subscribe`, (done) => {
+    const machine = createMachine({
+      id: 'machine',
+      initial: 'initial',
+      context: {
+        count: 0
+      },
+      states: {
+        initial: {
+          on: { activate: 'active' }
+        },
+        active: {}
+      }
+    });
+
+    const nextSpy = jest.fn().mockImplementation(() => {
+      throw new Error(
+        'doesnt_notify_error_listener_when_error_is_thrown_in_subscribe'
+      );
+    });
+    const errorSpy = jest.fn();
+
+    const actor = interpret(machine).start();
+
+    actor.subscribe({
+      next: nextSpy,
+      error: errorSpy
+    });
+    actor.send({ type: 'activate' });
+
+    expect(nextSpy).toHaveBeenCalledTimes(1);
+    expect(errorSpy).toHaveBeenCalledTimes(0);
+
+    installGlobalOnErrorHandler((ev) => {
+      ev.preventDefault();
+      expect(ev.error.message).toEqual(
+        'doesnt_notify_error_listener_when_error_is_thrown_in_subscribe'
+      );
+      setTimeout(() => {
+        done();
+      }, 2000);
```

Comment:
Should it be this long?

Reply:
actually, this is just some debugging leftover - it should just call `done()` right away